### PR TITLE
require edit_data role to access report builder

### PR DESCRIPTION
**Pull request by calellowitz**
@NoahCarnahan @czue @emord @millerdev
http://manage.dimagi.com/default.asp?222275#1115130

### CODE REVIEW

File: corehq/apps/userreports/views.py [hunk 1]
```diff
@@ -138,6 +139,7 @@ def create_report(request, domain):
 
 class ReportBuilderView(BaseDomainView):
 
+    @method_decorator(require_permission(Permissions.edit_data))
     @cls_to_view_login_and_domain
```
Comment (millerdev) [inline corehq/apps/userreports/views.py]: Is this redundant with `Permissions.edit_data`?

Comment (calellowitz) [inline corehq/apps/userreports/views.py]: I am not quite sure I understand your question. This uses the existing `Permissions.edit_data` to restrict access to Report builder to web users that have the edit data permission

Comment (millerdev) [inline corehq/apps/userreports/views.py]: Sorry for not asking more clearly. I was wondering if `cls_to_view_login_and_domain` was also doing a permission check at a lower level (which would be redundant and unnecessary), but looking at it again I think it's probably fine.

Comment (calellowitz) [inline corehq/apps/userreports/views.py]: I see now, I believe the `cls_to_view_login_and_domain` checks for user authentication and domain access, without checking what type of user role that web user has in the domain. My addition adds the role check

**Pull request merged**